### ISSUE TITLE
Size of SDK is now too large for our App Clip

### ISSUE THREAD

**Issue by robertsammons** [stripe/stripe-ios#1931]
The latest Stripe update appears to have added a full 1mb to the file size.

I can see that there is now the 1mb App Clip Apple Pay integration but this isn't suitable for our needs. 

Is there any further steps that can be taken to reduce the size of the SDK?

**Comment by csabol-stripe** [comment]
Hi @robertsammons, the additional 1mb was added due to the new ApplePay framework as well as the merge of a large feature branch for Link. We are actively planning ways to modularize the SDK more to support App Clips. Could you share which other components of the SDK would be needed to support your use case?

**Comment by robertsammons** [comment]
@csabol-stripe Hey Cameron, we use card payments as well as Apple Pay within our App Clip with full custom UI. I wonder if maybe imagery could be separated out, not sure what the size of all the card types and such would be ?

**Comment by davidme-stripe** [comment]
We're about to launch iOS SDK 23.0.0, which will split the Payments SDK into five modules. You can mix and match these to embed only the components you need.
* [StripePayments](https://github.com/stripe/stripe-ios/tree/beta/23.0.0-1/StripePayments) contains the API bindings.
* [StripePaymentsUI](https://github.com/stripe/stripe-ios/tree/beta/23.0.0-1/StripePaymentsUI) includes StripePayments, STPPaymentCardTextField, and other UI elements. 
* [StripeApplePay](https://github.com/stripe/stripe-ios/tree/beta/23.0.0-1/StripeApplePay) contains support for Apple Pay.
* [Stripe](https://github.com/stripe/stripe-ios/tree/beta/23.0.0-1/) includes all of the above packages and our legacy integrations.
* [StripePaymentSheet](https://github.com/stripe/stripe-ios/tree/beta/23.0.0-1/StripePaymentSheet) contains our main PaymentSheet UI.

Here are the App Store sizes for each. (Note that this is the *total* size added by the module, including all dependencies. For example, `StripePaymentsUI` shares many components with `StripeApplePay`, so embedding both would only add an incremental ~100-200kb over embedding `StripePaymentsUI` alone.)

```
+----------------------------+-----------------+-------------------+
| Module                     | Compressed size | Uncompressed size |
+----------------------------+-----------------+-------------------+
|                     Stripe |          2247kb |            5099kb |
|             StripePayments |          1335kb |            3207kb |
|           StripePaymentsUI |          1707kb |            3916kb |
|         StripePaymentSheet |          2748kb |            6254kb |
|             StripeApplePay |           425kb |            1018kb |
+----------------------------+-----------------+-------------------+
```

This is an improvement from v22 (3MB compressed, 6.9MB uncompressed), but it's still bigger than we'd like. We'll continue to explore ways to reduce the size of the SDK.

If you'd like to try it out, point Swift Package Manager or your Podfile at the `beta/23.0.0-1` branch. Thanks!

**Comment by davidme-stripe** [comment]
We're about to split the SDK into separate modules, which will allow you to embed only the card text field and Apple Pay. More details and a beta link here: https://github.com/stripe/stripe-ios/issues/1820#issuecomment-1279365907

**Comment by davidme-stripe** [comment]
We released this today as [23.0.0](https://github.com/stripe/stripe-ios/releases/tag/23.0.0). We've also posted a [migration guide](https://stripe.com/docs/mobile/ios/sdk-23-migration?lang-ios=swift) with module size information and tips for migrating to the latest version.

**Comment by robertsammons** [comment]
Thanks for this, we've been able to reduce our App Clip down by 1.8mb leaving us plenty of buffer room now
